### PR TITLE
Add USE_OPENSSL_QUIC for cmake

### LINKED ISSUE #12988: Missing cmake functionality for compile http3 with openssl and nghttp3
Missing cmake functionality for compile http3 with openssl and nghttp:
This feature is done here: #12734 only in autoconf without support cmake functionality. 
Please implement cmake functionality for this option.

### PR DESCRIPTION
**Pull request by talregev**
Fix #12988
Add USE_OPENSSL_QUIC for cmake

### CODE REVIEW

**Comment by bagder** [comment]
Maybe mention this build method in the `HTTP3.md` documentation?

**Review by vszakats — APPROVED** [review]

**Comment by talregev** [comment]
> Maybe mention this build method in the `HTTP3.md` documentation?

Done.

File: docs/HTTP3.md [hunk 1]
```diff
@@ -233,6 +233,14 @@ Build curl:
      % make
      % make install
 
+You can build curl with cmake:
```
Comment (vszakats) [inline docs/HTTP3.md:236]: Let's add an empty line here to match existing entries and make this render a code block:
```suggestion
You can build curl with cmake:

```

https://github.com/curl/curl/blob/master/docs/HTTP3.md

Comment (talregev) [reply docs/HTTP3.md:236]: Done.

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**